internal/pkg/template/input/input.go

```diff
@@ -73,16 +73,17 @@ type Value struct {
 }
 
 type Input struct {
-	Id          string      `json:"id" validate:"required,template-input-id"`
-	Name        string      `json:"name" validate:"required,min=1,max=25"`
-	Description string      `json:"description" validate:"max=60"`
-	Type        Type        `json:"type" validate:"required,template-input-type,template-input-type-for-kind"`
-	Kind        Kind        `json:"kind" validate:"required,template-input-kind"`
-	Default     interface{} `json:"default,omitempty" validate:"omitempty,template-input-default-value,template-input-default-options"`
-	Rules       Rules       `json:"rules,omitempty" validate:"omitempty,template-input-rules"`
-	If          If          `json:"showIf,omitempty" validate:"omitempty,template-input-if"`
-	Options     Options     `json:"options,omitempty" validate:"template-input-options"`
-	ComponentId string      `json:"componentId,omitempty"`
+	Id           string      `json:"id" validate:"required,template-input-id"`
+	Name         string      `json:"name" validate:"required,min=1,max=25"`
+	Description  string      `json:"description" validate:"max=60"`
+	Type         Type        `json:"type" validate:"required,template-input-type,template-input-type-for-kind"`
+	Kind         Kind        `json:"kind" validate:"required,template-input-kind"`
+	Default      interface{} `json:"default,omitempty" validate:"omitempty,template-input-default-value,template-input-default-options"`
+	Rules        Rules       `json:"rules,omitempty" validate:"omitempty,template-input-rules"`
+	If           If          `json:"showIf,omitempty" validate:"omitempty,template-input-if"`
+	Options      Options     `json:"options,omitempty" validate:"template-input-options"`
+	ComponentId  string      `json:"componentId,omitempty" validate:"required_if=Kind oauth"`
```

Comment:
Pri `ComponentId` chybala validacia, pridane + test.

Reply:
A jo, zůstala mi tady zapomenutá: https://github.com/keboola/keboola-as-code/pull/738 (snažil jsem se tam dodat validaci na to aby oauth inputy nemohly být prázdný ale pak se mi nepodařilo oworkaroundovat že v CLI jsou prázdný by design.)

Reply:
jj, prazdny `{}` sa uz validuje tiez spravne, je to niekde v dalsich PRs. Naprv som tam pridal `required` do nejakeho CLI testu (to islo) a v dalsom kroku som fixol a otestovat validaciu cez API.